lib/site_prism/addressable_url_matcher.rb

```diff
@@ -66,14 +66,14 @@ def extract_component_templates
       COMPONENT_NAMES.each_with_object({}) do |component, component_templates|
         component_url = to_substituted_uri.public_send(component).to_s
 
-        next unless component_url && component_url != ''
+        next unless component_url && !component_url.empty?
```

Comment:
Is there ever a chance this is NOT a string (given the `.to_s` above)?   If not this could devolve to the simple `next if component_url.empty?`

Reply:
As far as I could tell from debugging (I spent a while on this), the tests sometimes have strings and sometimes don't. I didn't check which circumstances have which.

I think this ideally will serve as a "first refactor" with someone else with perhaps a bit more ability fixing this up